lib/travis/build/script/c.rb

```diff
@@ -43,7 +43,13 @@ def use_directory_cache?
         private
 
           def compiler
-            config[:compiler].to_s
+            case config[:compiler].to_s
+            when /^gcc/i, /^clang/i
+              config[:compiler].to_s
+            else
+              { 'linux' => 'gcc', 
```

Comment:
Your suggestion does not what I want to do :-) Maybe there's a prettier `begins_with` function in Ruby that does what I want, but essentially, I want to check if the string begins with gcc or clang. If so, then leave the suffix (the compiler version) intact, otherwise replace it with a platform-dependent default.